### ISSUE TITLE
Is calling `m.save_message()` prior to adding attachment the correct approach to send large size file attachments?

### ISSUE THREAD

**Issue by nachiketrss** [O365/python-o365#841]
Hello

I am exploring this package and tried the following code:

```
from O365 import Account

credentials = ('client_id', 'client_secret')

account = Account(credentials, auth_flow_type='credentials', tenant_id='my_tenant_id')
if account.authenticate():
    print('Authenticated!')
    mailbox = account.mailbox('sender_email@my_domain.com') 
    m = mailbox.new_message()
    m.to.add('to_example@example.com')
    m.subject = 'Testing!'
    m.body = "George Best quote: I've stopped drinking, but only while I'm asleep."
    m.attachment.add = 'filename.txt'
    m.send()
```

It works when file size is under 4MB. When the file size is beyond that value it gives HTTP 413 error.

To get around this issue I followed the documentation and tried using the `beta` protocol as follows:

```
from O365 import Account, MSGraphProtocol

credentials = ('client_id', 'client_secret')

protocol = MSGraphProtocol(api_version='beta')
account = Account(credentials, auth_flow_type='credentials', tenant_id='my_tenant_id', protocol=protocol)
if account.authenticate():
    print('Authenticated!')
    mailbox = account.mailbox('sender_email@my_domain.com') 
    m = mailbox.new_message()
    m.to.add('to_example@example.com')
    m.subject = 'Testing!'
    m.body = "George Best quote: I've stopped drinking, but only while I'm asleep."
    m.attachment.add = 'filename.txt'
    m.send()
```

The above is making use of `beta` protocol as mentioned in the documentation but it doesn't work (return 413 error/same as 1st example) for attachment file size beyond 4MB.

The following solution worked but since it is not documented I want to get some advise on this please:

 ```
from O365 import Account

credentials = ('client_id', 'client_secret')

account = Account(credentials, auth_flow_type='credentials', tenant_id='my_tenant_id')
if account.authenticate():
    print('Authenticated!')
    mailbox = account.mailbox('sender_email@my_domain.com') 
    m = mailbox.new_message()
    m.to.add('to_example@example.com')
    m.subject = 'Testing!'
    m.body = "George Best quote: I've stopped drinking, but only while I'm asleep."
    m.save_message()
    m.attachment.add = 'filename.txt'
    m.send()
```

So there are 2 issues:
1. Beta protocol isn't able to send mail with large (>4MB) attachments
2. Adding `m.save_message()` before `attachment.add` seems to allow me to send emails with large file size without using `beta` protocol. I suppose this works by saving the email into drafts and then adding the attachment to it, and send.
 
Does the documentation need updating please to mention that calling `m.save_message()` prior to adding attachment is the correct approach to send large size file attachments?

**Comment by janscas** [comment]
Thanks! I’ll update the docs and also clear the error messages

**Comment by liberamiguel** [comment]
I am not able to send an attachment with neither alternative (using beta protocol or the usual one).
For me `'Message' object has no attribute 'attachment'`, also
`authenticate() missing 1 required keyword-only argument: 'scopes'`

Could it be related with the version I'm using? 
I've tried with 2.0.15 and 2.0.0

**Comment by kwilsonmg** [comment]
@janscas I believe that this was addressed with PR #1014 ? If you agree then we probably have #337 to close as well.

**Comment by janscas** [comment]
> @janscas I believe that this was addressed with PR #1014 ? If you agree then we probably have #337 to close as well.

Yes. thanks
